### DISCUSSION TITLE
How to update an existing block by id

### DISCUSSION THREAD

**Discussion by ezsolu** [mariosimao/notion-sdk-php#373] [Q&A] [answered]
Hi,

I can't find any document about how to update an existing block by using it's id, follow this API https://developers.notion.com/reference/update-a-block

Does this library support this feature?

**Comment by mariosimao** [comment, marked as answer]
Hello @ezsolu. Sorry for the late reply.

Yes, this library supports updating a block by its ID. 

The following example should help you.

```php
<?php

use Notion\Notion;

$token = $_ENV["NOTION_SECRET"];
$notion = Notion::create($token);

$blockId = "c986d7b0-7051-4f18-b165-cc0b9503ffc2";
$block = $notion->blocks()->find($blockId);

// Update your block here...

$updatedBlock = $notion->blocks()->update($block);
```
